qubits/sawtooth_qubit.py

```diff
@@ -0,0 +1,183 @@
+    
+    settings:
+        pinw: pinw of CPW the notch attaches to
+        gapw: gapw of CPW the notch attaches to
+        taper_angle: inner angle of tapers on both sides of the notch
+        notch_type: currently only takes 'trapezoid'
+        leftright: if refjunc.direction is up, draws the notch on the 'left' or
+            'right' side of the CPW
+        refjunc: endpoint of CPW the notch attaches to
+        offset: if refjunc.direction is up, positive offset shifts the notch 
+            down along the CPW
+            
+        h_padding: distance between notch taper and paddle
+        v_padding: distance between notch wall and paddle
+        updown: "points" the paddles either in refjunc.direction ('up') or the 
+            reverse ('down')
+        tooth_height: height of a tooth
+        tooth_spacing: distance between consecutive teeth on opposite paddles.
+            (for example teeth on the same paddle are separated by
+             2*tooth_spacing+tooth_width)
+        tooth_width: width of a tooth
+        
+        paddle_length: length of entire paddle (including pocket)
+        paddle_width: width of paddle NOT INCLUDING tooth_height
+        buffer: length of paddle where there will be no teeth (and a pocket)
+        
+        pocket_offset: distance from edge of pocket to end of paddle
+        pocket_depth: depth of pocket
+        pocket_length: full length of pocket (not just the length of the
+            deepest part of the pocket)
+        pocket_wall_angle: angle the wall makes with the side of the paddle
+            that is "cut out"
+    
+    it is somewhat hard to describe these things, there are pictures in docs
+    """
+    _defaults = {}
+    _defaults['pinw'] = 20
+    _defaults['gapw'] = 8.372
+    _defaults['refjunc'] = None
+    _defaults['offset'] = 0
+    _defaults['pin_gap'] = 8.372
+    _defaults['leftright'] = 'right'
+    _defaults['taper_angle'] = 60
+    
+    _defaults['h_padding'] = 10
+    _defaults['v_padding'] = 20
+    _defaults['updown'] = 'up'
+    
+    _defaults['tooth_height'] = 16
+    _defaults['tooth_spacing'] = 8
+    _defaults['tooth_width'] = 8
+    _defaults['paddle_length'] = 600
+    _defaults['paddle_width'] = 16
+    _defaults['paddle_gap'] = 24
+    
+    _defaults['buffer'] = 24
+    _defaults['pocket_offset'] = 8
+    _defaults['pocket_depth'] = 8
+    _defaults['pocket_length'] = 16
+    _defaults['pocket_wall_angle'] = 60
+    
+    def __init__(self,structure,settings):
+        
+        s = structure
+        
+        comp_key = 'ExampleQubit'
+        global_keys = ['pinw','gapw','pin_gap']
+        object_keys = ['pinw','gapw','gapw'], # which correspond to the extract global_keys
+        Component.__init__(self,structure,comp_key,global_keys,object_keys,settings)
+        
+        coords = self.refjunc.coords
+        direction = self.refjunc.direction + 180
+        
+        if self.paddle_gap - self.tooth_height < 6:
```

Comment:
feels like there might be a way to generalize this check to n variables directly. We could write a parser or something that would raise an error automatically

Reply:
I struggled to figure this out, especially because some of the small features were a difference between variables

we could do it more generally without the parser

small_features = [self.property1-self.property2,self.property3]
for feature in small features
    if feature < 6:
        print('your feature is too tiny')